resources/js/data-access/memo-api/tquery/export.ts

```diff
@@ -0,0 +1,39 @@
+import {V1} from "../config";
+import {DataItem, DataRequest, DataResponse} from "./types";
+
+/** The number of table cells (columns * rows) to request on a single page. */
+const NUM_CELLS_PER_PAGE = 50000;
+
+/** Fetches all the rows from tquery and returns them asynchronously. */
+export async function* getAllRowsExportIterator({
+  entityURL,
+  baseRequest,
+}: {
+  entityURL: string;
+  baseRequest: DataRequest;
+}): AsyncIterator<DataItem> {
+  if (!baseRequest.columns.length) {
+    return;
+  }
+  const pageSize = Math.max(1, Math.ceil(NUM_CELLS_PER_PAGE / baseRequest.columns.length));
+  function fetchPage(offset: number) {
+    return V1.post<DataResponse>(`${entityURL}/tquery`, {
+      ...baseRequest,
+      paging: {offset, size: pageSize},
+    }).then((res) => res.data);
+  }
+  let offset = 0;
+  let resp: DataResponse;
+  let fetch = fetchPage(offset);
+  for (;;) {
```

Comment:
while(true)?? Wydaje mi się lepiej komunikować nieskończoną pętlę, ale może kwestia gustu.

Reply:
Mi się podoba `for(;;)` przez to jakie jest bez sensu :) Raczej każdy zna.